src/slack.py

```diff
@@ -0,0 +1,127 @@
+import json
+import os
+import random
+import requests
+import sys
+import time
+
+
+# Constants
+
+
+bearer_token = 'Bearer ' + os.getenv('API_KEY')
+IMAGE_URL = 'https://raw.githubusercontent.com/cuappdev/bagel-bot/master/bagel-logo.png'
+SLACK_API = 'https://slack.com/api/'
+TESTING_CHANNEL_ID = 'CTF81MFH6'
+
+
+# Pull / Get Request
+
+
+def pr(endpoint, data):
+    return requests.post(
+        SLACK_API + endpoint,
+        data = json.dumps(data),
+        headers = {
+            'Authorization': bearer_token,
+            'Content-type': 'application/json'
+        }
+    ).json()
+
+
+def gr(endpoint, params=None):
+    return requests.get(
+        SLACK_API + endpoint,
+        params = params,
+        headers = {
+            'Authorization': bearer_token,
+            'Content-type': 'application/x-www-form-urlencoded'
+        }
+    ).json()
```

Comment:
while clean, we might want to wrap the `.json()` in try-excepts in case slack apis are down or change spec or something

Reply:
What should the app do in the except block besides print a message to standard out?

Reply:
mmm actually im not so sure about this anymore. I just want to make sure that if something critical crashes, we prevent ourselves from making more requests without the correct context and mess something up

Reply:
In that sense, maybe the except should just throw a custom error with a message to force ourselves to exit the process.

Reply:
Doesn't an uncaught exception already do this though?

Reply:
shiiii nvm